### PR TITLE
Ijaegle pi0

### PR DESCRIPTION
**Pull request by igjaegle**
3 changes:
1/ Add as default value for the vertex fit the beam spot position in x & y
2/ Create new pi0 fcal skimmer that is saving TOF  (based on the old pi0 cal)
3/ Create new pi0 fcal analyzer (based on the old pi0 fcal analyzer)

### CODE REVIEW

**Comment by gluex** [comment]
Test status for this pull request: SUCCESS
 
  Summary: [/work/halld/pull_request_test/halld_recon^ijaegle_pi0/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle_pi0/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^ijaegle_pi0/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle_pi0/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle_pi0/make_ijaegle_pi0.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle_pi0/make_ijaegle_pi0.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle_pi0/report_ijaegle_pi0.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle_pi0/report_ijaegle_pi0.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle_pi0

File: src/plugins/Utilities/pi0fcaltofskim/JEventProcessor_pi0fcaltofskim.cc [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+
+      Double_t x = locFCALShowers[j]->getPosition().X();
+      Double_t y = locFCALShowers[j]->getPosition().Y();
+   //    Double_t z = locFCALShowers[j]->getPosition().Z() ;
+ //cout << "Z: " << z << endl;
+      //DVector3 pos_FCAL(x,y,625.406);
+      //for LH2 target
+      //DVector3 pos_FCAL(0,0,625.406);
+      
+      DVector3 pos_FCAL(0,0,638);
```
Comment (sdobbs) [inline src/plugins/Utilities/pi0fcaltofskim/JEventProcessor_pi0fcaltofskim.cc]: Why not get this from the XML geometry?

**Comment by sdobbs** [comment]
Looks generally reasonable - did you test that TOF data is getting properly written out? it would be good to check a few events - since this is a raw data format, the raw hits get written to the EVIO file, and the TOF points reconstructed from scratch when reading the skimmed file back in.

**Comment by sdobbs** [comment]
@staylorjlab , does the proposed change to DVertex_factory make sense?

**Comment by igjaegle** [comment]
Good point, I will take the FCAL position from the db. And, I did check that the TOF data is correcly written out.

File: src/plugins/Utilities/pi0fcaltofskim/JEventProcessor_pi0fcaltofskim.cc (hunk 1 above)
Comment (igjaegle) [reply src/plugins/Utilities/pi0fcaltofskim/JEventProcessor_pi0fcaltofskim.cc]: In fact, it is not used.

**Comment by igjaegle** [comment]
The only remaining issue is the question of Simon. If the fit is failing as in primex, the default values defined in  DVertex_factory are given.

**Comment by staylorjlab** [comment]
We seen for most of our running time that it appeared that the center of the beam spot  in x has been off by about 2 mm (you can see it in the vertex monitoring plots), but this was not taken into account before in the vertex factory, so I'm okay with this change to DVertex.  

The DVertex factory reports the target center in z if the track fitting code fails to fit any tracks (so it uses the default values).   In the monitoring_hists histogram for the "vertex" (EventVertexZ)  for all such events this shows up as a spike.

**Pull request merged**